File: backend/src/main/java/com/gpuoj/controller/ProblemController.java
package com.gpuoj.controller;

import com.gpuoj.model.Problem;
import com.gpuoj.service.ProblemService;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

@RestController
@RequestMapping("/api/problems")
@RequiredArgsConstructor
public class ProblemController {

    private final ProblemService problemService;

    @GetMapping
    public Flux<Problem> listProblems() {
        return problemService.listProblems();
    }

    @GetMapping("/{id}")
    public Mono<Problem> getProblem(@PathVariable String id) {
        return problemService.getProblem(id);
    }
}

File: backend/src/main/kotlin/com/gpuoj/controller/ProblemController.kt
package com.gpuoj.controller

import com.gpuoj.model.Problem
import com.gpuoj.service.ProblemService
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono

@RestController
@RequestMapping("/api/problems")
class ProblemController(private val problemService: ProblemService) {

    @GetMapping
    fun listProblems(): Flux<Problem> = problemService.listProblems()

    @GetMapping("/{id}")
    fun getProblem(@PathVariable id: String): Mono<Problem> = problemService.getProblem(id)
}
